From 83d936acc1c8472053c079d87ef47f0f63f115c7 Mon Sep 17 00:00:00 2001
From: Hou Tao <houtao1@huawei.com>
Date: Mon, 23 May 2022 21:18:15 +0800
Subject: [PATCH 2956/4585] [Backport] xfs: add trace for read and release of
 regular file

hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I53R0H

Reference: https://gitee.com/openeuler/kernel/commit/2ae705d8329adf3431710ca16a37c47d446cd225

--------------------------------

Use fs_file_read_do_trace() and trace_fs_file_release() to do that.

Signed-off-by: Hou Tao <houtao1@huawei.com>
Acked-by: fang wei <fangwei1@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Zhihao Cheng <chengzhihao1@huawei.com>
Reviewed-by: Zhang Yi <yi.zhang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>

Signed-off-by: Wang Hai <wanghai38@huawei.com>
---
 fs/xfs/xfs_file.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fs/xfs/xfs_file.c b/fs/xfs/xfs_file.c
index 2fb8cf556f8d..80adec66744b 100644
--- a/fs/xfs/xfs_file.c
+++ b/fs/xfs/xfs_file.c
@@ -29,6 +29,7 @@
 #include <linux/backing-dev.h>
 #include <linux/mman.h>
 #include <linux/fadvise.h>
+#include <trace/events/fs.h>
 
 static const struct vm_operations_struct xfs_file_vm_ops;
 
@@ -289,6 +290,7 @@ xfs_file_buffered_aio_read(
 	ssize_t			ret;
 
 	trace_xfs_file_buffered_read(ip, iov_iter_count(to), iocb->ki_pos);
+	fs_file_read_do_trace(iocb);
 
 	if (iocb->ki_flags & IOCB_NOWAIT) {
 		if (!xfs_ilock_nowait(ip, XFS_IOLOCK_SHARED))
@@ -1197,6 +1199,7 @@ xfs_file_release(
 	struct inode	*inode,
 	struct file	*filp)
 {
+	trace_fs_file_release(inode, filp);
 	return xfs_release(XFS_I(inode));
 }
 
-- 
2.17.1

